### PR TITLE
[AP-3662] Add stylesheet linting

### PR DESCRIPTION
**Pull request by naseberry**
## What

[Add stylesheet linting](https://dsdmoj.atlassian.net/browse/AP-3662)

Add stylesheet linter and configure it using GOV.UK standards to make the code consistent and  follows the GOV.UK standards.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/webpack/stylesheets/admin.scss [hunk 1]
```diff
@@ -1,29 +1,27 @@
 .admin-page {
-  nav.app-navigation {
+  .app-navigation {
     @media (min-width: 40.0625em) {
```
Comment (naseberry) [inline app/webpack/stylesheets/admin.scss:3]: Clean as you code (CAYC): we could use the design system's media queries throughout our stylesheets.

File: app/webpack/stylesheets/admin.scss [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
+    padding: 0;
+  }
 
-        a {
-          padding: 0.5em 2em;
-          background-color: #f3f2f1;
-        }
+  .app-navigation .app-navigation__list li a {
+    padding: .5em 2em;
+    background-color: #f3f2f1;
```
Comment (naseberry) [inline app/webpack/stylesheets/admin.scss:20]: CAYC: we could use the design system's color mixin throughout our stylesheets.

File: app/webpack/stylesheets/bank-selection.scss [hunk 3]
```diff
@@ -1,11 +1,11 @@
 .bank-selection {
   .govuk-radios__item {
     border: 1px solid #0b0c0c;
+    background: #ffffff; // Old browsers
+    background: -moz-linear-gradient(left, #ffffff 0%, #ffffff 31%, #ffffff 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 100%); // FF3.6-15
+    background: -webkit-linear-gradient(left, #ffffff 0%, #ffffff 31%, #ffffff 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 100%); // Chrome10-25,Safari5.1-6
+    background: linear-gradient(to right, #ffffff 0%, #ffffff 31%, #ffffff 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 31%, #f8f8f8 100%); // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
```
Comment (naseberry) [inline app/webpack/stylesheets/bank-selection.scss:7]: CAYC: we could use autoprefixer

File: app/webpack/stylesheets/check-your-answers.scss [hunk 4]
```diff
@@ -58,16 +48,16 @@ $govuk-fonts-path: "~govuk-frontend/govuk/assets/fonts/";
 
 .govuk-summary-list__total-row {
   @include govuk-media-query($until: tablet) {
-    //border-top: 3px solid #000;
-    //margin-bottom: govuk-spacing(3);
+    // border-top: 3px solid #000;
+    // margin-bottom: govuk-spacing(3);
```
Comment (naseberry) [inline app/webpack/stylesheets/check-your-answers.scss:52]: TODO: all property commented, can remove the media query

File: app/webpack/stylesheets/govuk-mods.scss [hunk 5]
```diff
@@ -1,24 +1,31 @@
...
 .govuk-panel {
   dl.govuk-list {
-    color: white;
+    color: govuk-colour("white");
     font-size: inherit;
   }
 }
 
-//Start button IE11 fix - adding a width of 15.5px to the SVG (AP-933)
+// Start button IE11 fix - adding a width of 15.5px to the SVG (AP-933)
```
Comment (naseberry) [inline app/webpack/stylesheets/govuk-mods.scss:26]: CAYC: older browsers no longer supported, we should remove.

File: .overcommit.yml [hunk 6]
```diff
@@ -29,6 +29,10 @@ PreCommit:
     command: ['yarn', 'run', 'standard']
     requires_files: true
     include: 'app/webpack/**/*.js'
+  Stylelint:
+    enabled: true
+    command: ['yarn', 'run', 'stylelint']
+    requires_files: true
```
Comment (colinbruce) [inline .overcommit.yml:35]: :wave: Just an observation from when I added the db anonymizer step below...
I believe that by setting requires_files you can set an include block that will make the task _only_ run if those files are changed.

With this setup, the step will run on **every** commit regardless of whether the S/CSS has been changed.

If you add 
```yaml
include: 'app/webpack/stylesheets/**/*.scss'
```
overcommit will only run when the scss files are changed and make it slightly faster :+1:

File: app/webpack/stylesheets/admin.scss (hunk 1 above)
Comment (naseberry) [reply app/webpack/stylesheets/admin.scss:3]: To be actioned by https://dsdmoj.atlassian.net/browse/AP-3699

File: app/webpack/stylesheets/check-your-answers.scss (hunk 4 above)
Comment (naseberry) [reply app/webpack/stylesheets/check-your-answers.scss]: Sorted.

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

A regression test of the admin pages might be useful as the `.nav` changes seem a little off.

Also I couldn't see a need for disabling `selector-max-id` but that is probably CSS 'magic' that isn't obvious to me! :D

File: app/webpack/stylesheets/govuk-secondary-button.scss [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
   color: #0b0c0c;
+  background-color: govuk-colour("light-grey");
   box-shadow: 0 2px 0 #b5babe;
 }
 
-.govuk-button.govuk-secondary-button:active, 
-.govuk-button.govuk-secondary-button:hover, 
-.govuk-button.govuk-secondary-button:focus  {
-  background-color: #d0d3d6;
+.govuk-button.govuk-secondary-button:active,
```
Comment (cpjmcquillan) [inline app/webpack/stylesheets/govuk-secondary-button.scss:7]: I'm unsure why we need these styles at all.

I'd expect us to be using `govuk-button--secondary` as per the design system, so we may be able to remove this stylesheet and replace the usage of `govuk-secondary-button` in the `_next_action_button` partial.

**Review by cpjmcquillan — APPROVED** [review]
LGTM 🎖️ 

I noticed some of these stylesheets are redundant, so I've opened #4694 to remove them.

**Comment by naseberry** [comment]
> LGTM 👍
> 
> A regression test of the admin pages might be useful as the `.nav` changes seem a little off.
> 
> Also I couldn't see a need for disabling `selector-max-id` but that is probably CSS 'magic' that isn't obvious to me! :D

Cheers, I will UAT the admin pages.

No magic here, `selector-max-id` is to limit the used of ID selectors, it is to reduce specificity in what is a componentized system that is meant to be reusable.

File: app/webpack/stylesheets/govuk-secondary-button.scss (hunk 7 above)
Comment (naseberry) [reply app/webpack/stylesheets/govuk-secondary-button.scss:7]: I spotted a bit of this, therefore I create https://dsdmoj.atlassian.net/browse/AP-3699 to replace variables, functions and mixins.

**Pull request merged**